libafl/src/corpus/inmemory.rs

```diff
@@ -270,20 +217,102 @@ where
         self.map.iter().next_back().map(|x| *x.0)
     }
 
-    /// Create new `TestcaseStorage`
-    #[must_use]
-    pub fn new() -> Self {
+    fn new() -> Self {
         Self {
-            map: TestcaseStorageMap::default(),
-            keys: vec![],
-            progressive_idx: 0,
+            #[cfg(not(feature = "corpus_btreemap"))]
+            map: hashbrown::HashMap::default(),
+            #[cfg(feature = "corpus_btreemap")]
+            map: alloc::collections::BTreeMap::default(),
+            keys: Vec::default(),
             #[cfg(not(feature = "corpus_btreemap"))]
             first_idx: None,
             #[cfg(not(feature = "corpus_btreemap"))]
             last_idx: None,
         }
     }
 }
+/// Storage map for the testcases (used in `Corpus` implementations) with an incremental index
+#[derive(Default, Serialize, Deserialize, Clone, Debug)]
+#[serde(bound = "I: serde::de::DeserializeOwned")]
+pub struct TestcaseStorage<I>
+where
+    I: Input,
+{
+    /// The map in which testcases are stored
+    pub enabled: TestcaseStorageMap<I>,
+    pub disabled: TestcaseStorageMap<I>,
+    /// The progressive idx
+    progressive_idx: usize,
+}
+
+impl<I> UsesInput for TestcaseStorage<I>
+where
+    I: Input,
+{
+    type Input = I;
+}
+
+impl<I> TestcaseStorage<I>
+where
+    I: Input,
+{
+    /// Insert a testcase assigning a `CorpusId` to it
+    #[cfg(not(feature = "corpus_btreemap"))]
+    pub fn insert(&mut self, testcase: RefCell<Testcase<I>>, is_disabled: bool) -> CorpusId {
```

Comment:
We could have an `insert` and an `insert_disabled` to keep the overall API the same

Reply:
👍